locations/spiders/new_york_pizza_nl_de.py

```diff
@@ -0,0 +1,57 @@
+import re
+from typing import Iterable
+
+import scrapy
+from scrapy import Request, Spider
+from scrapy.http import Response
+
+from locations.dict_parser import DictParser
+from locations.hours import DAYS_DE, DAYS_EN, OpeningHours
+
+
+class NewYorkPizzaNLDESpider(Spider):
+    name = "new_york_pizza_nl_de"
+    item_attributes = {"brand": "New York Pizza", "brand_wikidata": "Q2639128"}
+
+    def start_requests(self) -> Iterable[Request]:
+        for country, page in (("nl", "vestigingen"), ("de", "filialen")):
+            yield scrapy.Request(
+                f"https://www.newyorkpizza.{country}/{page}",
+                callback=self.parse_auth_token,
+                cb_kwargs={"top_level_domain": country},
+            )
+
+    def parse_auth_token(self, response: Response, top_level_domain: str = None) -> Iterable[Request]:
+        auth_token = response.xpath('//input[@id="completeAntiForgeryToken"]/@value').get()
+        yield scrapy.FormRequest(
+            url=f"https://www.newyorkpizza.{top_level_domain}/General/GetFilteredStores/",
+            formdata={"includeSliceStores": "false", "storeIds": ""},
+            headers={"RequestVerificationToken": auth_token},
+            callback=self.parse,
+            cb_kwargs={"top_level_domain": top_level_domain},
+        )
+
+    def parse(self, response, top_level_domain=None, **kwargs):
+        for store in response.json():
+            item = DictParser.parse(store)
+            if item["name"].startswith("New York Pizza "):
+                item["branch"] = item["name"].lstrip("New York Pizza ")
```

Comment:
strip doesn't do what you think it does. See the serene example https://www.w3schools.com/python/ref_string_lstrip.asp


You want removeprefix